File: modules/app/src/main/java/org/artembogomolova/demo/webapp/security/LogoutHandlerImpl.java
package org.artembogomolova.demo.webapp.security;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import lombok.extern.slf4j.Slf4j;
import org.springframework.security.core.Authentication;
import org.springframework.security.web.authentication.logout.LogoutHandler;
import org.springframework.stereotype.Service;

@Service
@Slf4j
public class LogoutHandlerImpl implements LogoutHandler {

  @Override
  public void logout(HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse, Authentication authentication) {
    log.info("trying logout {}", authentication.getName());
  }
}

File: app/src/main/kotlin/org/artembogomolova/demo/webapp/main/security/LogoutHandlerImpl.kt
package org.artembogomolova.demo.webapp.main.security

import org.springframework.security.core.Authentication
import org.springframework.security.web.authentication.logout.LogoutHandler
import org.springframework.stereotype.Service
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse

@Service
class LogoutHandlerImpl : LogoutHandler {
    override fun logout(httpServletRequest: HttpServletRequest, httpServletResponse: HttpServletResponse, authentication: Authentication) {
        println("trying logout ${authentication.name}")
    }
}